components/motor/gpio/loop.go

```diff
@@ -164,6 +156,34 @@ func (m *EncodedMotor) createControlLoopConfig() control.Config {
 				DependsOn: []string{"endpoint"},
 			},
 		},
-		Frequency: 100,
+		Frequency: 100.0,
 	}
+
+	if p == 0.0 && i == 0.0 && d == 0.0 {
+		// when tuning (all PID are zero), the loop has to exclude the trapz block, so the sum block is
+		// temporarily changed to depend on the constant block instead of the trapz block to pass over trapz
+		sumBlock := control.BlockConfig{
+			Name: "sum",
+			Type: "sum",
+			Attribute: rdkutils.AttributeMap{
+				"sum_string": "+-",
+			},
+			DependsOn: []string{"set_point", "derivative"},
+		}
+		conf.Blocks = append(conf.Blocks, sumBlock)
+	} else {
+		// if not tuning, set the sum block to once again depend
+		// on the trapz block instead of the constant bloc
+		sumBlock := control.BlockConfig{
+			Name: "sum",
+			Type: "sum",
+			Attribute: rdkutils.AttributeMap{
+				"sum_string": "+-",
+			},
+			DependsOn: []string{"trapz", "derivative"},
+		}
+		conf.Blocks = append(conf.Blocks, sumBlock)
+	}
```

Comment:
We can go ahead and try to make this a helper function from the package. Might help with semantically calling that function from outside packages. What do you think?

Okay to do in a follow up PR.

Reply:
+1 , this portion can be a function.